### ISSUE TITLE
adjacent-overload-signatures false positives with static methods

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#1768]
### Bug Report

- __TSLint version__: `4.0.1`
- __TypeScript version__: `2.0.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Foo {
  public static bar() { ... }

  constructor() { ... }

  public bar() { ... }
  ~~~~~~~~~~~~~~~~~~~~ adjacent-overload-signatures
}
```

#### Actual behavior

Lint failure expecting the instance method `bar` and static method of the same name to be adjacent to each other.

#### Expected behavior

Do not flag this as a failure.
